src/azure/msal/azureAccountProvider.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+
+import { AccountInfo, PublicClientApplication } from '@azure/msal-node';
+import { AzureAuthType, AzureResource, IAccountKey, IAccountProvider, IAccountProviderMetadata, IPromptFailedResult } from '../../models/contracts/azure';
+import { IToken, MsalAzureAuth } from './msalAzureAuth';
+import { IDeferred } from '../../models/interfaces';
+import { getAzureActiveDirectoryConfig } from '../utils';
+import { MsalAzureCodeGrant } from './msalAzureCodeGrant';
+import { MsalAzureDeviceCode } from './msalAzureDeviceCode';
+import VscodeWrapper from '../../controllers/vscodeWrapper';
+import { Logger } from '../../models/logger';
+import { IAccount } from 'vscode-mssql';
+import * as AzureConstants from '../constants';
+import * as LocalizedConstants from '../../constants/localizedConstants';
+
+export class AzureAccountProvider implements IAccountProvider {
+
+	private readonly authMappings = new Map<AzureAuthType, MsalAzureAuth>();
+	private initComplete!: IDeferred<void, Error>;
+	private initCompletePromise: Promise<void> = new Promise<void>((resolve, reject) => this.initComplete = { resolve, reject });
+	public clientApplication: PublicClientApplication;
+	public vscodeWrapper: VscodeWrapper;
+	public logger: Logger;
+
+	constructor(
+		metadata: IAccountProviderMetadata,
+		context: vscode.ExtensionContext,
+		clientApplication: PublicClientApplication,
+		vscodeWrapper: VscodeWrapper,
+		logger: Logger
+	) {
+		this.clientApplication = clientApplication;
+		this.vscodeWrapper = vscodeWrapper;
+		this.logger = logger;
+
+		vscode.workspace.onDidChangeConfiguration((changeEvent) => {
```

Comment:
I'm not sure why this class would be needed.. since we only have MSAL now, what is the purpose of moving code here?

Reply:
You mean azureAccountProvider? I created it to handle the mappings of different auth providers such as public cloud, china cloud, gov cloud

Reply:
Had a chat with Cheena offline, will be refactoring this to be inside msalAzureController